### PR TITLE
added data attribute for contentlet element

### LINKED ISSUE #24217: Add number of pages in the contentlet HTML
### Parent Issue

#23223

### User Story

As a developer, I want to be able to know if a contentlet is being used on more than one page.

We need to ask the user if he would like to edit a contentlet just for the page it's being edited, or for all the pages that use that contentlet.

The page builder uses `data-attr` to get all the information about a contentlet or container (identifier, inode, lang, etc.), we need to add the number of pages as well.

### Acceptance Criteria

- [ ] Add `data-dot-on-number-of-pages="N"` to the `<div data-dot-object="contentlet">` element.

### Proposed Objective

Please Select

### Proposed Priority

Please Select

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

http://localhost:8080/api/v1/page/render/test?language_id=1&mode=EDIT_MODE

In this enpdoint you will get the "rendered" property is where the HTML comes.

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by hassan-mustafa-baig**
### Proposed Changes
* Added a new data attribute for the contentlet element

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

image

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java [hunk 1]
```diff
@@ -15,13 +15,16 @@
 import com.dotmarketing.portlets.containers.business.FileAssetContainerUtil;
 import com.dotmarketing.portlets.containers.model.Container;
 import com.dotmarketing.portlets.containers.model.FileAssetContainer;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
 import com.dotmarketing.portlets.fileassets.business.FileAsset;
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java:20]: I think these import are not used

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java [hunk 2]
```diff
@@ -15,13 +15,16 @@
...
 import com.dotmarketing.portlets.folders.model.Folder;
 import com.dotmarketing.portlets.templates.design.bean.ContainerUUID;
 import com.dotmarketing.util.Logger;
 import com.dotmarketing.util.PageMode;
 import com.dotmarketing.util.UtilMethods;
 import com.liferay.util.StringPool;
+import io.vavr.control.Try;
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java:27]: I think this import are not used

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java [hunk 3]
```diff
@@ -318,6 +321,8 @@ private InputStream buildVelocity(final Container container, final String uuid,
                     velocityCodeBuilder.append("<div")
                         .append(" data-dot-object=")
                         .append("\"contentlet\"")
+                        .append(" data-dot-on-number-of-pages=")
+                        .append("\"$ON_NUMBER_OF_PAGES\"")
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java:325]: are you sure that you need both?, I mean are you sure that you need the velocity variable and the data attribute too?

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java (hunk 3 above)
Comment (hassan-mustafa-baig) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java:325]: yes we need it. Please check the related ticket https://github.com/dotCMS/core/issues/24217 thanks

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java (hunk 2 above)
Comment (hassan-mustafa-baig) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java]: yes you are right. it has been addressed - thanks

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java (hunk 1 above)
Comment (hassan-mustafa-baig) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java]: this has been addressed - thanks

**Review by jcastro-dotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentletLoader.java [hunk 4]
```diff
@@ -115,6 +116,9 @@ public InputStream buildVelocity(final Contentlet content, final PageMode mode,
         sb.append("#set($CONTENT_TYPE_ID='").append(contentTypeId).append("' )");
         sb.append("#set($CONTENT_LANGUAGE='").append(content.getLanguageId()).append("' )");
 
+        final Optional<Integer> pageReferences =
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentletLoader.java:119]: we need to check if it is on the content map, also if it is not on the map, you can only get the references here if the mode is EDIT MODE

Comment (hassan-mustafa-baig) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentletLoader.java]: This feedback has been addressed - thanks

**Review by jdotcms — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

**Pull request merged**